File: packages/rook_sdk_health_connect/android/src/main/kotlin/com/rookmotion/rook_sdk_health_connect/mapper/EnvironmentMappers.kt
package com.rookmotion.rook_sdk_health_connect.mapper

import com.rookmotion.rook.sdk.domain.environment.RookEnvironment
import com.rookmotion.rook_sdk_health_connect.proto.EnvironmentProto

fun EnvironmentProto.toRookEnvironment(): RookEnvironment {
    return if (this == EnvironmentProto.SANDBOX) {
        RookEnvironment.SANDBOX
    } else {
        RookEnvironment.PRODUCTION
    }
}

File: packages/rook_sdk_health_connect/lib/src/data/mapper/environment_mappers.dart
import 'package:rook_sdk_core/rook_sdk_core.dart';
import 'package:rook_sdk_health_connect/src/data/proto/protos.pbenum.dart';

extension RookEnvironmentMappers on RookEnvironment {
  EnvironmentProto toProto() {
    return switch (this) {
      RookEnvironment.sandbox => EnvironmentProto.SANDBOX,
      _ => EnvironmentProto.PRODUCTION,
    };
  }
}
